### PR TITLE
Doc using pydata-sphinx-theme

### PR DESCRIPTION
**Pull request by tupui**
Part of #438

Note that I already went ahead for the branch name in the links and also the repo address.

The following will be fixed later on with #436
```
.../rbc/doc/omnisci/array.rst: WARNING: document isn't included in any toctree
.../rbc/doc/omnisci/bytes.rst: WARNING: document isn't included in any toctree
.../rbc/doc/omnisci/column.rst: WARNING: document isn't included in any toctree
.../rbc/doc/omnisci/columnlist.rst: WARNING: document isn't included in any toctree
.../rbc/doc/omnisci/outputcolumn.rst: WARNING: document isn't included in any toctree
```

### CODE REVIEW

**Review by guilhermeleobas — APPROVED** [review]
LGTM.

Not related to this PR, but I just notice some signatures have the typingctx as first parameter. I need to remove this from showing up in the docs.

image

File: doc/conf.py [hunk 1]
```diff
@@ -84,19 +80,20 @@
...
-    'analytics_id': '',
-    'logo_only': False,
-    'display_version': True,
-    'prev_next_buttons_location': 'bottom',
-    # Toc options
-    'collapse_navigation': True,
-    'sticky_navigation': True,
-    # 'navigation_depth': 4,
+    "logo_link": "index",
+    "github_url": "https://github.com/quansight/rbc",
```
Comment (pearu) [inline doc/conf.py:90]: just a note: this does not exist yet

File: test_requirements.txt [hunk 2]
```diff
@@ -6,3 +6,4 @@ thriftpy2
 six
 netifaces
 sphinx_autodoc_typehints
+pydata-sphinx-theme>=0.8.0
```
Comment (pearu) [inline test_requirements.txt:9]: This contradicts https://github.com/xnd-project/rbc/pull/436 but I guess pydata-sphinx-theme is what we want. Right?

File: doc/conf.py [hunk 3]
```diff
@@ -14,14 +14,14 @@
...
-import datetime
+from datetime import date
 from rbc import __version__
 
 
 # -- Project information -----------------------------------------------------
 
 project = 'RBC'
-copyright = '2018-%s, Xnd-Project Developers' % datetime.datetime.now().year
+copyright = f'2018-{date.today().year}, Xnd-Project Developers'
```
Comment (pearu) [inline doc/conf.py:24]:
```suggestion
copyright = f'2018-{date.today().year}, Quansight RBC Developers'
```

**Review by pearu — APPROVED** [review]
I have a suggestion and a couple of comments. Otherwise, LGTM! Thanks, @tupui!

File: test_requirements.txt (hunk 2 above)
Comment (tupui) [reply test_requirements.txt:9]: Yes this is what we want here.

File: doc/conf.py (hunk 1 above)
Comment (tupui) [reply doc/conf.py:90]: Yeah I will revert this (and bellow for the `html_context`) quick so we can merge and I can change it as soon as we moved the repo. Same for the logo actually, I suppose we should put Quansight's one once we move.

When we do the move we should also take the opportunity to rename the branch to `main`. BTW, who is doing this? I don't think I have permissions to help on that.

Comment (tupui) [inline doc/conf.py:90]:
```suggestion
    "github_url": "https://github.com/xnd-project/rbc",
```

File: doc/conf.py [hunk 4]
```diff
@@ -84,19 +80,20 @@
...
+    "github_url": "https://github.com/quansight/rbc",
+    "use_edit_page_button": True,
 }
 
-# The Xnd logo
-html_logo = "images/xndlogo.png"
+html_context = {
+    "github_user": "quansight",
+    "github_repo": "rbc",
+    "github_version": "main",
```
Comment (tupui) [inline doc/conf.py:97]:
```suggestion
    "github_user": "xnd-project",
    "github_repo": "rbc",
    "github_version": "master",
```

**Review by tupui — COMMENTED** [review]
Thanks @pearu!

File: doc/conf.py (hunk 1 above)
Comment (pearu) [reply doc/conf.py]: Before moving, let's have one rbc release first.
^ @guilhermeleobas 
Re main branch, see https://github.com/xnd-project/rbc/issues/328

**Pull request merged**